src/ipfs/schemas.ts

```diff
@@ -0,0 +1,101 @@
+import { z } from 'zod';
+
+const VERSION_1 = '1.0' as const;
+const VERSION_2 = '2.0' as const;
+
+// Ethereum address regex (case insensitive)
+const ethereumAddressRegex = /^0x[a-f0-9]{40}$/i;
+
+export const coverValidatorsSchema = z.object({
+  version: z.literal(VERSION_1),
+  validators: z
+    .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+    .min(1, 'At least one validator address is required'),
+});
+
+export const coverQuotaShareSchema = z.object({
+  version: z.literal(VERSION_1),
+  quotaShare: z.number().min(0).max(100),
+});
+
+export const coverAumCoverAmountPercentageSchema = z.object({
+  version: z.literal(VERSION_1),
+  aumCoverAmountPercentage: z.number().min(0).max(100),
+});
+
+export const coverWalletAddressSchema = z.object({
+  version: z.literal(VERSION_1),
+  walletAddress: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+});
+
+export const coverWalletAddressesSchema = z.discriminatedUnion('version', [
+  z.object({
+    version: z.literal(VERSION_1),
+    walletAddresses: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+  }),
+  z.object({
+    version: z.literal(VERSION_2),
+    walletAddresses: z
+      .array(z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'))
+      .min(1, 'At least one wallet address is required'),
+  }),
+]);
+
+export const coverFreeTextSchema = z.object({
+  version: z.literal(VERSION_1),
+  freeText: z.string().min(1, 'Free text cannot be empty'),
+});
+
+export const coverDesignatedWalletsSchema = z.object({
+  version: z.literal(VERSION_1),
+  wallets: z
+    .array(
+      z.object({
+        wallet: z.string().regex(ethereumAddressRegex, 'Invalid Ethereum address'),
+        amount: z.string().regex(/^\d+$/, 'Amount must be a valid wei value'),
```

Comment:
are these usdc denominated only?

Reply:
Not sure. @mixplore do you know?

This is `coverDesignatedWalletsSchema` used by open-cover from what I remember from our conversation

Reply:
No, this is used for Base Defi Pass product, which can be denominated in ETH, USDC or cbBTC. UI for reference: https://v2.nexusmutual.io/cover/product/227

So far, FE doesn't have any extra validation for this - this is used only for the OpenCover team and as far as I remember, when we launched it we agreed to a simple csv file format with a wallet address and an amount (string) => FE converts it to a json array of objects with `wallet` and `amount` string fields. If we decide to add a more thorough validation, I should adjust FE as well to match that.

Reply:
@rackstar I just had a look at existing covers with this type of metadata: they have `wallet`, `amount` AND `currency` all as string values. So let's add `currency` as well here

Reply:
updated as discussed: https://github.com/NexusMutual/sdk/pull/295/files/6d3423e39dea052e0784bf45659d66bcfc449b0a..608e65585b15a3929860c3fb7bc7c8542da81afb